spras/config/heuristics.py

```diff
@@ -0,0 +1,105 @@
+import os
+
+import networkx as nx
+from pydantic import BaseModel, ConfigDict
+
+from spras.interval import Interval
+from spras.statistics import compute_statistics, statistics_options
+
+all = ['GraphHeuristicsError', 'GraphHeuristic']
+
+class GraphHeuristicsError(RuntimeError):
+    """
+    Represents an error arising from a graph algorithm output
+    not meeting the necessary graph heuristisc.
+    """
+    failed_heuristics: list[tuple[str, float | int, list[Interval]]]
+
+    @staticmethod
+    def format_failed_heuristic(heuristic: tuple[str, float | int, list[Interval]]) -> str:
+        name, desired, intervals = heuristic
+        if len(intervals) == 1:
+            interval_string = str(intervals[0])
+        else:
+            formatted_intervals = ", ".join([str(interval) for interval in intervals])
+            interval_string = f"one of the intervals ({formatted_intervals})"
+        return f"{name} expected {desired} in interval {interval_string}"
```

Comment:
This text doesn't quite match up. You could get "in interval one of the intervals..."